components/map/basic/src/leaflet/shapes/Polygons.js

```diff
@@ -19,62 +20,45 @@ export default class SearchMapPolygons {
     }
   }
 
-  highlightFeature(evt) {
-    const layer = evt.target
-
-    layer.setStyle({
-      color: '#666',
-      dashArray: '',
-      fillOpacity: 0.7,
-      weight: 5
-    })
-
-    if (!L.Browser.ie && !L.Browser.opera) {
-      layer.bringToFront()
-    }
-  }
-
-  resetHighlight(evt) {
-    if (this._geojson !== null) {
-      this._geojson.resetStyle(evt.target)
-    }
-  }
-
-  zoomIn(evt) {
+  static zoomIn(evt) {
     const map = evt.target._map
     map.setView(evt.latlng, map.getZoom() + 2)
   }
 
-  onEachFeature(feature, layer) {
-    layer.on({
-      dblclick: SearchMapPolygons.zoomIn,
-      mouseout: SearchMapPolygons.resetHighlight,
-      mouseover: SearchMapPolygons.highlightFeature
-    })
-  }
-
   printPolygonOnMap({map, polygon}) {
-    const params = {
+    const {hoverStyles} = this
+    const polygonGeoJSon = L.geoJson(polygon, {
       className: 'scm-map__area',
-      onEachFeature: SearchMapPolygons.onEachFeature
-    }
-
-    const polygonGeoJSon = L.geoJson(polygon, params)
+      onEachFeature: (feature, layer) => {
+        layer.on({
+          mouseout: function() {
+            polygonGeoJSon.resetStyle(this)
+          },
+          mouseover: function() {
+            this.setStyle(hoverStyles)
+
+            if (!L.Browser.ie && !L.Browser.opera) {
```

Comment:
Why this?

Reply:
tbh, don't know, it was already in the code and I've moved to another line ;)